packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -77,15 +78,32 @@ class CupertinoNavigationBar extends StatelessWidget implements PreferredSizeWid
 
   @override
   Widget build(BuildContext context) {
-    Widget styledMiddle = middle;
-    if (styledMiddle.runtimeType == Text || styledMiddle.runtimeType == DefaultTextStyle) {
-      // Let the middle be black rather than `actionsForegroundColor` in case
-      // it's a plain text title.
-      styledMiddle = DefaultTextStyle.merge(
-        style: const TextStyle(color: CupertinoColors.black),
-        child: middle,
-      );
-    }
+    final TextStyle actionsStyle = leading == null && trailing == null ? null : new TextStyle(
+      fontSize: 17.0,
+      letterSpacing: -0.24,
+      color: actionsForegroundColor,
+    );
+
+    final Widget styledLeading = leading == null ? null : DefaultTextStyle.merge(
+      style: actionsStyle,
+      child: leading,
+    );
+
+    final Widget styledTrailing = trailing == null ? null : DefaultTextStyle.merge(
+      style: actionsStyle,
+      child: trailing,
+    );
+
+    // Let the middle be black rather than `actionsForegroundColor` in case
+    // it's a plain text title.
+    final Widget styledMiddle = middle == null ? middle : DefaultTextStyle.merge(
+      style: const TextStyle(
```

Comment:
what about just actionsStyle.copyWith so everything except the color is in one place?